java/core/src/test/org/apache/orc/impl/TestRecordReaderImpl.java

```diff
@@ -2449,4 +2449,23 @@ public void testSargApplier() throws Exception {
     f2.setAccessible(true);
     assertFalse((boolean)f2.get(applier1));
   }
+
+  @Test
+  public void testRgEndOffset() throws IOException {
+    Configuration conf = new Configuration();
+    Path filePath = new Path(getClass().getClassLoader().
+        getSystemResource("orc-small-compression-size.orc").getPath());
```

Comment:
Do you think we can create this file in this test case, @hsnusonic ?

Reply:
Good point. I can try to generate files for the test.